src/main/java/io/supertokens/totp/Totp.java

```diff
@@ -0,0 +1,54 @@
+package io.supertokens.totp;
+
+import java.io.IOException;
+
+import io.supertokens.Main;
+import io.supertokens.pluginInterface.totp.TOTPDevice;
+import io.supertokens.pluginInterface.totp.sqlStorage.TOTPSQLStorage;
+import io.supertokens.storageLayer.StorageLayer;
+
+public class Totp {
+
+    public static CreateDeviceResponse createDevice(Main main, String userId, String deviceName, int skew, int period)
+            throws IOException {
+
+        TOTPSQLStorage totpStorage = StorageLayer.getTOTPStorage(main);
+        String secret = GenerateDeviceSecret.generate();
+
+        if (userId == null || deviceName == null || secret == null) {
+            throw new IllegalArgumentException("userId, deviceName and secret cannot be null");
+        }
+
+        try {
+            TOTPDevice device = new TOTPDevice(userId, deviceName, secret, skew, period, false);
+            totpStorage.createDevice(device);
+        } catch (Exception e) {
+            throw new IOException(e);
+        }
+
+        return new CreateDeviceResponse("deviceName", secret);
+    }
+
+    private static class GenerateDeviceSecret {
+        // private final String secret;
+
+        // private GenerateDeviceSecret(String secret) {
+        // this.secret = secret;
+        // }
+
+        public static String generate() {
```

Comment:
if this class only has this one static function, then no need to wrap it in a class. Just define this function in the Totp class itself as a static function

Reply:
Removed.